satpy/enhancements/__init__.py

```diff
@@ -120,6 +128,65 @@ def func(band_data, index=None):
     return apply_enhancement(img.data, func, separate=True)
 
 
+def interp_scaling(
```

Comment:
I'm not really found of that name, it feels cryptic. How about `curve_scaling` instead?

(by the way, would this allow for higher order interpolation?)

Reply:
`interp` obviously comes from the numpy name so I thought it was more obvious this way. I'm not a fan of `curve` as to me that sounds more like spline fitting or something. I could fit `linear` somewhere in there?

Reply:
How about `piecewise_linear_stretch`?

Reply:
I do like the consistency of the `stretch` name and I am a fan of piecewise as a term in general. :wink:

Reply:
sounds good with `piecewise_linear_stretch`!